src/components/modal/v2/lib/hooks/currency.js

```diff
@@ -10,6 +10,7 @@ export const currencyFormat = string => {
             // eslint-disable-next-line security/detect-unsafe-regex
             /((\$|£)?(\d{1,5}(\.|,)){1,3}00)/g
         ) ?? null;
+
     if (match !== null) {
         match.forEach(foundString => {
             const filteredString = foundString
```

Comment:
```suggestion
            const filteredString = foundString
                .replace(/(\.|,)00-/g, '-')
                .replace(/(\.|,)00$/g, '')
                .replace(/(\.|,)00(.|\s*)EUR/g, ' €');
            formattedStr = formattedStr.replace(foundString, filteredString);
```
We should be good to simplify this regex to the above since we aren't dealing with messages here.